components/src/button/button.directive.ts

```diff
@@ -60,7 +60,7 @@ export class ButtonDirective {
       "focus:tw-ring-primary-700",
     ]
       .concat(this.block ? ["tw-w-full", "tw-block"] : ["tw-inline-block"])
-      .concat(buttonStyles[this.buttonType ?? "secondary"]);
+      .concat(buttonStyles[this.buttonType] ?? []);
```

Comment:
But will `this.buttonType` ever be null if it has a default value? Can't we just do

```suggestion
      .concat(buttonStyles[this.buttonType]);
```

?

If I'm missing something and we do need to account for `this.buttonType` being null, then your original code would be better - my query was whether we need the `??` at all.

Reply:
I think so. `[].concat(undefined) => [undefined]`. The new code is logically better since it ensures we don't keep `undefined` in the array.